base/version.jl

```diff
@@ -299,7 +299,7 @@ function banner(io::IO = stdout)
            _       _ _(_)_     |  Documentation: https://docs.julialang.org
           (_)     | (_) (_)    |
            _ _   _| |_  __ _   |  Type \"?\" for help, \"]?\" for Pkg help.
-          | | | | | | |/ _` |  |
+          | | | | | | |/ _` |  |  Type \"exit()\" to exit.
           | | |_| | | | (_| |  |  Version $(VERSION)$(commit_date)
```

Comment:
Seems a bit weird to give documentation its own line, gap, then a big block of text covering help and version, etc. 

If including this is necessary, and it deserves a whole line, then maybe the "paragraph break" should be between documentation-like things and version-like things. Or maybe the version-like things should move down a line, to preserve space.

Reply:
Yeah, I'm hesitant to add a whole new line just for an `exit` hint.

Reply:
I'd prefer we do https://github.com/JuliaLang/julia/pull/38522 (or rather, @mkitti's suggested alternative to https://github.com/JuliaLang/julia/pull/38522) instead of adding a whole new line to the banner.

That's why I tried to sneak it into an existing line, but since that made the line too long, I'd probably prefer that we do something like https://github.com/JuliaLang/julia/pull/38522 instead of this PR.

Reply:
+1 vote for the clever error hints, instructions at the point of need.

Reply:
Yeah that's what I'm thinking.

Plus, the banner already links to the docs, and the getting started page of the docs (

Reply:
Instead of adding this to the banner, what if we add this hint to the output when you type `?` and press enter.

The banner already says: "Type "?" for help, "]?" for Pkg help."

So, when the user types `?` and presses enter, let's put the "type `exit()` to close the REPL" hint at the bottom of that help message.

That, plus a hint for if the user types `exit` (no parentheses) and presses enter, and we're probably good.

Reply:
Yes that message is another possibly useful place. Although to be fair I never found it until some thread here about upgrading it :/

Reply:
Here's that PR: https://github.com/JuliaLang/julia/pull/41997